osclib/stagingapi.py

```diff
@@ -1088,27 +1088,43 @@ def map_ring_package_to_subject(self, project, pkg):
 
         return project
 
-    def get_sub_packages(self, package, project=None):
+    def get_sub_packages(self, package, project):
         """
-        Returns a list of packages that need to be linked into rings
-        too. A package is actually a tuple of project and package name
+        Returns a list of packages that need to be linked into main package.
+        For adi package, check specfiles according to the main package.
         """
         ret = []
-        if not project:
+
+        # Started the logic. Note that, return empty tuple in case selecting
+        # non-ring package to a letter staging.
+        if self.is_adi_project(project):
+            # The additional check ensure the package is exist
```

Comment:
nit: `item_exists(project, package)` is prety clear that checking if the package exists in project without comment.